### PR TITLE
Exp34 Terraform Code, Exp35 Pipeline Destroy Terraform, Exp36 Pipeline Apply Terraform

### PR DESCRIPTION
**Pull request by manuelanut**
Tasks Exp34 Terraform Code, Exp35 Pipeline Destroy Teraform, Exp36 Pipeline Apply Terraform
-Add Terraform code for creating Azure Resources, add Terraform Pipeline Apply and Terraform Pipeline Destroy
All 3 files were tested

### CODE REVIEW

File: .github/workflows/pipeline destroy terraform.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Terraform Destroy
+
```
Comment (en-vasiliuralucaelena) [inline .github/workflows/pipeline destroy terraform.yml:2]: Please do not use empty spaces in file names

File: .github/workflows/pipeline destroy terraform.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+on:
+  workflow_dispatch:
+   
+jobs:
+  terraform-workflow:
+    runs-on: ubuntu-latest
+    name: Terraform Workflow
+    env:
+         ARM_CLIENT_ID: ${{ secrets.TF_CLIENT_ID }}
```
Comment (en-nader-said) [inline .github/workflows/pipeline destroy terraform.yml:11]: Correct indentation pls, 2 spaces after env

File: terraform/.gitignore [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+operations/automation-script/variable.json
+operations/automation-script/workspace.template.json
+operations/automation-script/workspace.json
+operations/sentinel-policies-scripts/create-policy.template.json
+operations/sentinel-policies-scripts/create-policy.json
+operations/variable-scripts/variable.template.json
+operations/variable-scripts/variable.json
+
+Sentinel runtime directory
+.sentinel
```
Comment (en-nader-said) [inline terraform/.gitignore:43]: These are not used, delete them

File: terraform/.gitignore [hunk 4]
```diff
@@ -0,0 +1,43 @@
+Compiled files
+*.tfstate
+*.tfstate.backup
+*.tfstate.lock.info
+
+logs
+*.log
```
Comment (en-nader-said) [inline terraform/.gitignore:7]: Should remove this

File: terraform/.gitignore [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+Directories
+.terraform/
+.vagrant/
+
+SSH Keys
+*.pem
+
+Backup files
+*.bak
```
Comment (en-nader-said) [inline terraform/.gitignore:17]: Should remove this

File: terraform/mysql_db.tf [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  name                = "exp-db-mysql"
+  location            = data.azurerm_resource_group.exp_rg.location
+  resource_group_name = data.azurerm_resource_group.exp_rg.name
+  administrator_login          = var.administrator_login
+  administrator_login_password = var.administrator_login_password
+  version             = "5.7"
+  sku_name   = "B_Gen5_2"
+  storage_mb = 5120
+
+  auto_grow_enabled                 = true
```
Comment (en-nader-said) [inline terraform/mysql_db.tf:11]: Set this to false please

File: terraform/provider.tf [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+}
+
+provider "azurerm" {
+  features {}
+
+  #client_id       = var.client_id
+  #client_secret   = var.client_secret
+  #tenant_id       = var.tenant_id
+  #subscription_id = var.subscription_id
```
Comment (en-nader-said) [inline terraform/provider.tf:24]: Remove commented lines

File: terraform/variable.tf [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+#variable "tenant_id" {
+ # type      = string
+  #sensitive = true
+#}
+
+#variable "subscription_id" {
+  #type      = string
+  #sensitive = true
+#}
```
Comment (en-nader-said) [inline terraform/variable.tf:43]: Remove commented lines

File: .github/workflows/pipeline destroy terraform.yml [hunk 9]
```diff
@@ -0,0 +1,38 @@
+name: Terraform Destroy
+
+on:
+  workflow_dispatch:
```
Comment (en-vasiliuralucaelena) [inline .github/workflows/pipeline destroy terraform.yml:4]: You do not need the ':' here, please see https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

File: .github/workflows/pipeline-apply-terraform.yml [hunk 10]
```diff
@@ -0,0 +1,51 @@
+name: Terraform Apply
+
+on:
+  workflow_dispatch:
```
Comment (en-vasiliuralucaelena) [inline .github/workflows/pipeline-apply-terraform.yml:4]: You do not need the ':' here, please see https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

File: terraform/mysql_db.tf [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+  backup_retention_days             = 7
+  geo_redundant_backup_enabled      = false
+  infrastructure_encryption_enabled = false
+  public_network_access_enabled     = true
+  ssl_enforcement_enabled           = true
+  ssl_minimal_tls_version_enforced  = "TLS1_2"
+
```
Comment (en-vasiliuralucaelena) [inline terraform/mysql_db.tf:18]: Please run terraform fmt for all files, this file does not look right

File: terraform/provider.tf (hunk 7 above)
Comment (en-vasiliuralucaelena) [inline terraform/provider.tf:24]: The commented lines can be removed

File: terraform/variable.tf (hunk 8 above)
Comment (en-vasiliuralucaelena) [inline terraform/variable.tf:43]: These can be removed too

File: terraform/vm_front.tf [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+  admin_ssh_key {
+    username   = "azureuser"
+    public_key = var.frontend_ssh_key
+  }
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+  //doar os
```
Comment (en-vasiliuralucaelena) [inline terraform/vm_front.tf:22]: Please have only English comments in the code

**Review by en-vasiliuralucaelena — CHANGES_REQUESTED** [review]

File: terraform/vm_back.tf [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+}
+
+resource "azurerm_network_interface" "exp_nic_backend" {
+  name                = "exp-nic-backend"
+  resource_group_name = data.azurerm_resource_group.exp_rg.name
+  location            = data.azurerm_resource_group.exp_rg.location
+
+  ip_configuration {
+    name                          = "testconfiguration2"
```
Comment (en-nader-said) [inline terraform/vm_back.tf:38]: Rename this to something other than test stuff

File: terraform/vm_front.tf [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+  }
+}
+
+resource "azurerm_network_interface" "exp_nic_frontend" {
+  name                = "exp-nic-frontend"
+  resource_group_name = data.azurerm_resource_group.exp_rg.name
+  location            = data.azurerm_resource_group.exp_rg.location
+
+  ip_configuration {
+    name                          = "testconfiguration2"
```
Comment (en-nader-said) [inline terraform/vm_front.tf:37]: Rename to something other than test stuff

**Review by en-nader-said — CHANGES_REQUESTED** [review]

**Review by en-nader-said — CHANGES_REQUESTED** [review]

**Review by en-vasiliuralucaelena — APPROVED** [review]

**Pull request merged**